api/src/main/java/org/openmrs/module/m2sysbiometrics/client/M2SysV105Client.java

```diff
@@ -84,17 +88,21 @@ public BiometricSubject enroll(BiometricSubject subject) {
 
     @Override
     public BiometricSubject update(BiometricSubject subject) {
-        M2SysCaptureResponse capture = scanDoubleFingers();
+        M2SysCaptureResponse fingerScan = scanDoubleFingers();
+        FingerScanStatus fingerScanStatus = checkIfFingerScanExists(fingerScan);
 
-        String response = localBioServerClient.update(subject.getSubjectId(), capture.getTemplateData());
-        M2SysResults results = XmlResultUtil.parse(response);
+        updateService.updateLocally(subject, fingerScan);
 
-        if (!results.isUpdateSuccess()) {
-            throw new M2SysBiometricsException("Unable to update fingerprints for: "
-                    + subject.getSubjectId());
+        String nationalId;
+        if (fingerScanStatus.isRegisteredNationally()) {
+            nationalId = fingerScanStatus.getNationalBiometricSubject().getSubjectId();
+            updateService.updateNationally(nationalId, fingerScan);
+        } else {
+            nationalId = nationalUuidGenerator.generate();
+            registrationService.registerNationally(nationalId, fingerScan);
```

Comment:
The generate method uses also nationalClient, which could have i.e. incorrectly set credentials. If we want to persist this failure in the retry queue, we should keep it mind. I think we should move id generating to registrationService.

Reply:
Right, in Oskar's PR it is moved so I won't change it for registration, but I will do it for update